src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -2187,9 +2186,9 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 		this.updateDropFeedback(tabsContainer, false, e, targetTabIndex);
 		tabsContainer.classList.remove('scroll');
 
-		let targetEditorIndex = this.tabsModel instanceof UnstickyEditorGroupModel ? targetTabIndex + this.groupView.stickyCount : targetTabIndex;
+		let targetEditorIndex = this.tabsModel.stickyCount === 0 ? targetTabIndex + this.groupView.stickyCount : targetTabIndex;
 		const options: IEditorOptions = {
-			sticky: this.tabsModel instanceof StickyEditorGroupModel && this.tabsModel.stickyCount === targetEditorIndex,
+			sticky: this.tabsModel.stickyCount > 0 && this.tabsModel.stickyCount === targetEditorIndex,
```

Comment:
StickyEditorGroupModel can also have a stickyCount of 0, so this check is not good.